code/__HELPERS/icon_smoothing.dm

```diff
@@ -240,29 +240,34 @@
 
 	var/list/New = list()
 
+	var/list/cached_overlays = A.our_overlays
+	var/init_len = LAZYLEN(cached_overlays)
+
 	if(A.top_left_corner != nw)
-		A.overlays -= A.top_left_corner
+		LAZYREMOVE(cached_overlays, A.top_left_corner)
```

Comment:
you can't use this for this, all overlay operations HAVE to go thru the overlay proc you make for the reasons i detailed above.

it would be better to keep a list of overlays to remove and then pass that to cut_overlay.